### PR TITLE
error metrics for filer and store

### PR DESCRIPTION
**Pull request by kmlebedev**

### CODE REVIEW

File: weed/server/filer_server_handlers_read.go [hunk 1]
```diff
@@ -182,6 +184,7 @@ func (fs *FilerServer) GetOrHeadHandler(w http.ResponseWriter, r *http.Request)
 
 		err = filer.StreamContent(fs.filer.MasterClient, writer, chunks, offset, size)
 		if err != nil {
+			stats.FilerRequestCounter.WithLabelValues("stream.contentFailed").Inc()
```
Comment (chrislusf) [inline weed/server/filer_server_handlers_read.go]: change to `read.stream.failed`?

File: weed/server/filer_server_handlers_read.go [hunk 2]
```diff
@@ -173,6 +174,7 @@ func (fs *FilerServer) GetOrHeadHandler(w http.ResponseWriter, r *http.Request)
 				Directory: dir,
 				Name:      name,
 			}); err != nil {
+				stats.FilerRequestCounter.WithLabelValues("read.cachefailed").Inc()
```
Comment (chrislusf) [inline weed/server/filer_server_handlers_read.go]: `read.cache.failed`?

File: weed/server/filer_server_handlers_read.go [hunk 3]
```diff
@@ -162,6 +162,7 @@ func (fs *FilerServer) GetOrHeadHandler(w http.ResponseWriter, r *http.Request)
 		if offset+size <= int64(len(entry.Content)) {
 			_, err := writer.Write(entry.Content[offset : offset+size])
 			if err != nil {
+				stats.FilerRequestCounter.WithLabelValues("write.entryfailed").Inc()
```
Comment (chrislusf) [inline weed/server/filer_server_handlers_read.go]: `write.entry.failed`?

File: weed/storage/needle/needle_read_write.go [hunk 4]
```diff
@@ -278,6 +283,7 @@ func (n *Needle) readNeedleDataVersion2(bytes []byte) (err error) {
 		n.NameSize = uint8(bytes[index])
 		index = index + 1
 		if int(n.NameSize)+index > lenBytes {
+			stats.VolumeServerRequestCounter.WithLabelValues("errorIndexOutOfRange").Inc()
```
Comment (zerospiel) [inline weed/storage/needle/needle_read_write.go:286]: Maybe it's better to move those strings such `errorIndexOutOfRange`, `errorCRC` etc. to a new block with constants to escape unnecessary repeats?

Same for `weed/topology` package

Comment (kmlebedev) [reply weed/storage/needle/needle_read_write.go:286]: Seems like a pretty rare event.
and more reason to go and see the logs and code

Comment (chrislusf) [reply weed/storage/needle/needle_read_write.go:286]: > Maybe it's better to move those strings such `errorIndexOutOfRange`, `errorCRC` etc. to a new block with constants to escape unnecessary repeats?
> 
> 
> 
> Same for `weed/topology` package

Good suggestion! It'll be easier to have consistent metrics names also.

**Comment by chrislusf** [comment]
@kmlebedev I merged the PR and refactored the error metric names to a separate file.

https://github.com/chrislusf/seaweedfs/commit/433fde4b186a8244aabeb8659bb901a89b56e213

This file contains metric names for all errors. Currently it is only refactoring.
But I think the naming convention can be `ErrorSomeThing = "error.some.thing"`.

Since you are actively working on this, I want to check with you first.

**Pull request merged**